kuma/javascript/src/bcd.js

```diff
@@ -271,45 +271,722 @@ export function activateBCDSignals(
+    };
+
+    /**
+     * Hides bc signal blocks and returns it to initial state
+     */
+    const finishReport = () => {
+        // Hide both form block and complete block
+        bcSignalBlock.classList.remove('open');
+        bcSignalCompleteBlock.classList.remove('open');
+
+        // Go to first step, so if will be shown if the block is opened again
+        toStep(1);
+        resetControls();
+    };
+
+    /**
+     * Sends all the data from controls to server, then shows signal complete block
+     */
+    const sendReport = () => {
+        const browsers = [];
+        for (const browser of document.querySelectorAll(
+            '.browser.selected > .browser-name'
+        )) {
+            browsers.push(browser.innerText);
+        }
+
+        const selectRow = document.getElementById('select-row');
+
+        let feature = '';
+        if (
+            selectRow &&
+            selectRow instanceof HTMLSelectElement &&
+            selectRow.options[selectRow.selectedIndex] &&
+            selectRow.options[selectRow.selectedIndex] instanceof
+                HTMLOptionElement &&
+            selectRow.options[selectRow.selectedIndex].value
+        ) {
+            feature = selectRow.options[selectRow.selectedIndex].value;
+        }
 
+        let explanation = '';
+        const explanationEl = document.getElementById('brief-explanation');
+        if (
+            explanationEl instanceof HTMLTextAreaElement &&
+            explanationEl.value
+        ) {
+            explanation = explanationEl.value;
+        }
+
+        let supportingMaterial = '';
+        const supportingMaterialEl = document.getElementById(
+            'supporting-material'
+        );
+        if (
+            supportingMaterialEl instanceof HTMLTextAreaElement &&
+            supportingMaterialEl.value
+        ) {
+            supportingMaterial = supportingMaterialEl.value;
+        }
+
+        const signalApiUrl = '/api/v1/bc-signal';
         const payload = {
             slug: slug,
-            locale: locale
+            locale: locale,
+            feature: feature,
+            explanation: explanation,
+            browsers: browsers.join(', '),
+            supporting_material: supportingMaterial // eslint-disable-line camelcase
         };
 
-        signalLink.addEventListener('click', function() {
-            fetch(signalApiUrl, {
-                method: 'POST',
-                body: JSON.stringify(payload),
-                headers: {
-                    'X-CSRFToken': getCookie('csrftoken'),
-                    'Content-Type': 'application/json'
+        bcSignalBlock.classList.remove('open');
+        bcSignalCompleteBlock.classList.add('open');
+
+        fetch(signalApiUrl, {
+            method: 'POST',
+            body: JSON.stringify(payload),
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken'),
+                'Content-Type': 'application/json'
+            }
+        });
```

Comment:
The final page "lies" :)
It said "Report Sent" when in fact the XHR post failed. (it failed on an 500 internal server error because my local mysql doesn't have the right database migration).

Essentially, I think we need to take care of that better. Getting a 500 from the server isn't completely unrealistic but what's very realistic is network failures or 504 gateway timeout or something.

Reply:
Hello @peterbe, thank you for the comment here.
I will design a new screen to cover this scenario. When we say network failure, what would the user's actions be in this event? Would they be able to "try again" or would this likely be a permanent "no can do"? would there be more than one common "fail" state?  
Thank you!

Reply:
My preferred pattern is:
1. Submit is triggered, disable the submit button, and send the `fetch()` request
2. If it all worked, fine, do what we we already have
3. if it failed (either exception caught or `!response.ok), enable the submit button again, and display a message in red that the server submission failed.

Reply:
@peterbe something like this?:
![Signalling - Step 2 error](https://user-images.githubusercontent.com/42435754/65336581-8dc0c200-dbbe-11e9-8dfa-ed0f213fe579.png)

Reply:
We can discuss the messaging when you turn that into code.